lib/components/shorts/shorts_videos_stream_player.dart

```diff
@@ -0,0 +1,154 @@
+import 'dart:async';
+
+import 'package:brunstadtv_app/components/loading_generic.dart';
+
+import 'package:brunstadtv_app/providers/shorts_videos_service.dart';
+import 'package:brunstadtv_app/components/shorts/shorts_video_player_screen.dart';
+import 'package:flutter/services.dart';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:video_player/video_player.dart';
+
+class ShortsVideosStreamPlayer extends ConsumerStatefulWidget {
+  const ShortsVideosStreamPlayer({
+    Key? key,
+    required this.shortsVideos,
+  }) : super(key: key);
+
+  final List<ShortsVideo> shortsVideos;
+
+  @override
+  ConsumerState<ShortsVideosStreamPlayer> createState() => _ShortsVideosStreamPlayerState();
+}
+
+class _ShortsVideosStreamPlayerState extends ConsumerState<ShortsVideosStreamPlayer> {
+  final Map<int, VideoPlayerController> _controllers = {};
+  Future<bool>? _future;
+  int prevPageIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _future = loadFirstVideo();
+  }
+
+  @override
+  void dispose() {
+    disposeControllers();
+    SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp]);
+    super.dispose();
+  }
+
+  void disposeControllers() {
+    List<int> indexList = [];
+    _controllers.forEach((key, value) {
+      indexList.add(key);
+      value.dispose();
+    });
+    _controllers.clear();
+  }
+
+  Future<bool> loadFirstVideo() async {
+    await _initializeControllerAtIndex(0);
+    await _initializeControllerAtIndex(1);
+    _playControllerAtIndex(0);
+    return true;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        PageView.builder(
+            scrollDirection: Axis.vertical,
+            itemCount: widget.shortsVideos.length,
+            onPageChanged: (pageIndex) async {
+              if (pageIndex > prevPageIndex) {
+                await _playNext(pageIndex);
+              } else {
+                await _playPrevious(pageIndex);
+              }
+              setState(() => prevPageIndex = pageIndex);
```

Comment:
you may need a `if(!mounted) return;` before the setState, because the widget could have been disposed while awaiting the operations above.

Reply:
The rule is that when you are going to do something after some async work, you always need to check if the widget is still `mounted` before using setState or the `context`.